src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,99 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.RegistrationException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String DEFAULT_USER_LOGIN_ONE = "DummyUser";
+    private static final String DEFAULT_USER_LOGIN_TWO = "DummyUserTwo";
+    private static final String DEFAULT_USER_PASSWORD = "DummyUserPassword";
+    private static final String SHORT_LOGIN = "Login";
+    private static final String SHORT_PASSWORD = "pass";
+    private static final int DEFAULT_USER_AGE = 18;
+    private StorageDao storageDao = new StorageDaoImpl();
+    private RegistrationService registrationService = new RegistrationServiceImpl();
+    private User dummyUser;
+
+    @BeforeEach
+    void setUp() {
+        dummyUser = new User();
+        dummyUser.setLogin(DEFAULT_USER_LOGIN_ONE);
+        dummyUser.setPassword(DEFAULT_USER_PASSWORD);
+        dummyUser.setAge(DEFAULT_USER_AGE);
+    }
+
+    @Test
+    void register_alreadyExists_notOk() {
+        storageDao.add(dummyUser);
+        RegistrationException exception = assertThrows(RegistrationException.class,
+                () -> registrationService.register(dummyUser));
+        assertEquals("Login already exists", exception.getMessage());
```

Comment:
The test `register_alreadyExists_notOk` assumes the `register` method checks for existing users, but this logic is not visible in the tested class. Ensure that the `RegistrationServiceImpl` has logic to handle duplicate user registration and throw an appropriate exception.